dbt_style_guide.md

```diff
@@ -1,237 +1,415 @@
-        tests:
-          - unique
-          - not_null
-
-      - name: event_time
-        description: "When the event occurred in UTC (eg. 2018-01-01 12:00:00)"
-        tests:
-          - not_null
-
-      - name: user_id
-        description: The ID of the user who recorded the event
-        tests:
-          - not_null
-          - relationships:
-              to: ref('users')
-              field: id
-```
+- Every subdirectory contains their own `.yml` file(s) which contain configurations for the models within the subdirectory.
+
+- A separate `.yml` file is created per top-level configuration (`sources`, `models`) that applies to the models within the subdirectory. 
+
+- YAML files should be prefixed with an underscore to keep it at the top of the subdirectory.  
+  Example: `__sources_core.yml`, `_models_core.yml`
+
+- YAML files should be named for the top-level configuration it contains.  
+  Examples: `_sources_core.yml`, `_models_core.yml`
+
+- Indents should use four spaces.
+
+- List items should be indented.
+
+- Use a new line to separate list items that are dictionaries, where appropriate.
+
+- Lines of YAML should be no longer than 80 characters.
+
+- ### Example YAML
+  ```yaml
+  version: 2
+
+  models:
+    - name: stg_snowplow__events
+      description: This model contains events from the core Jaffle Shop website.
+      columns:
+        - name: event_id
+          description: "{{ doc('event_id_description') }}"
+          tests:
+            - unique
+            - not_null
+
+        - name: event_at
+          description: When the event occurred in UTC (eg. 2018-01-01 12:00:00)
+          tests:
+            - not_null
+
+        - name: user_id
+          description: >
+            The user id of the visitor to the site.
+            This only populates when the user logs in - this user id can be  
+            joined to the Jaffle Shop Users data.
+          tests:
+            - not_null
+            - relationships:
+                to: ref('users')
+                field: id
+  ```
 
 
 ## Jinja style guide
 
-* When using Jinja delimiters, use spaces on the inside of your delimiter, like `{{ this }}` instead of `{{this}}`
-* Use newlines to visually indicate logical blocks of Jinja
+- Jinja delimiters should have spaces inside of the delimiter between the brackets and your code.  
+  Example: `{{ this }}` instead of `{{this}}`
+
+- Use [whitespace control](https://jinja.palletsprojects.com/en/3.1.x/templates/#whitespace-control) to make compiled SQL more readable.
+
+- An effort should be made for a good balance in readability for both templated 
+and compiled code. However, opt for code readability over compiled SQL readability
+when needed.
+
+- A macro file should be named after the _main_ macro it contains.
+
+- A file with more than one macro which will be used independently should follow
+these conventions:
+  - The macros are all related to a main idea 
+  - The file is named for the main idea 

```

Comment:
Not sure if this is best practice, but seems like if it is a macro used by lots of other macros it should not be buried within another macro file. thoughts?
```suggestion
  - The file is named for the main idea 
  - The macros nested within are not used by other macros
```

Reply:
Agree - will change this tomorrow!

Reply:
Actually, I decided to do this since it's the last one. I elaborated a little bit - how is [this](https://github.com/dbt-labs/corp/pull/63/commits/3d5743a86ea8f16147374ef66db4bfbc6037ec8f)?